buildv2.py

```diff
@@ -0,0 +1,807 @@
+#!/usr/bin/env python
+
+"""The kernelci builder script."""
+
+# pylint: disable=invalid-name
+# pylint: disable=broad-except
+
+import ConfigParser
+import argparse
+import fnmatch
+import io
+import json
+import logging
+import multiprocessing
+import os
+import platform
+import re
+import requests
+import shutil
+import stat
+import subprocess
+import sys
+import tempfile
+import time
+
+from urlparse import urljoin
+
+
+BUILD_CONFIG_FILE = "~/.buildpy.cfg"
+BUILD_LOG_FILE_NAME = "build.log"
+CONFIG_PATH = "arch/{0.arch:s}/configs/{1:s}"
+BUILD_UPLOAD_PATH = \
+    "{0.job:s}/{0.git_describe:s}/{0.arch:s}-{0.defconfig_full:s}"
+KBUILD_OUTPUT_PREFIX = "build"
+KBUILD_OUTPUT = KBUILD_OUTPUT_PREFIX
+_, KCONFIG_TMPFILE = tempfile.mkstemp(prefix="kconfig-")
+
+# How many times to retry API actions.
+SEND_RETRIES = 3
+
+# List of architectures on their cross-compiler.
+CROSS_COMPILERS = {
+    "arm": "arm-linux-gnueabi-",
+    "arm64": "aarch64-linux-gnu-",
+    "i386": None,
+    "x86": None,
+    "x86_64": None,
+}
+
+# Default build architecture.
+DEFAULT_ARCH = "arm"
+# Default cross-compiler.
+CROSS_COMPILE = CROSS_COMPILERS[DEFAULT_ARCH]
+
+# List of environment variable that will be checked.
+# Names are converted to upper case when checking their presence.
+ENV_VARIABLES = [
+    "arch",
+    "cross_compile",
+    "kbuild_output",
+    "ccache_disable",
+    "ccache_dir",
+    "git_describe",
+    "tree_name"
+]
+
+# Values to take out of a git repository, with the command necessary to extract
+# them.
+GIT_VARIABLES_AND_COMMANDS = {
+    "git_commit": r"git log -n1 --format=%H",
+    "git_url": "git config --get remote.origin.url",
```

Comment:
This is broken in repos that don't have a remote named 'origin'.

If you look at the original build.py, this command has a trailing "| cat", which is a hack so that the return is always a pass, and that git_url will be NULL in that case, but that's fine for local building in custom repos etc.